lm_eval/base.py

```diff
@@ -740,6 +741,9 @@ def process_results(self, doc, results):
                     rouge_scores = utils.flatten(rouge_scores)
                     # Merge all the rouge-type scores into the `out` dict.
                     out = {**out, **rouge_scores}
+                elif metric == "SARI":
+                    text = self.doc_to_text(doc)
+                    out = {"sari": metrics.sari(text, pred, target)}
```

Comment:
```suggestion
                    out["sari"] = metrics.sari(text, pred, target)
```
was overwriting all of out, so only sari was left.

Reply:
@cjlovering, Ok cool. Agree with the changes—didn't realise you could change it for all the prompts there... 

I made a few edits, as there were a few bugs when running again.
Also need to change
```
CONFIGURED_GENERATION_PS_METRICS = set(["BLEU", "ROUGE"])
```
to include SARI

Reply:
Okay, I think I addressed the smaller changes you mention + added a doc_to_rawtext attribute in `base.py:PromptSourceTask`